pkg/csi/controller.go

```diff
@@ -10,45 +18,132 @@ import (
 )
 
 type controllerServer struct {
+	config *controllerServerConfig
 }
 
 type controllerServerConfig struct {
+	driver *Driver
 }
 
 func newControllerServer(config *controllerServerConfig) csi.ControllerServer {
-	return &controllerServer{}
+	return &controllerServer{
+		config: config,
+	}
 }
 
 func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
-	return nil, nil
+	s.config.driver.config.Logger.Info("received create volume",
+		zap.String("name", req.Name))
+	// Check if volume==repo is there
+	repo := req.GetParameters()["repo"]
+	if repo == "" {
+		s.config.driver.config.Logger.Error("Repo not set", zap.String("name", req.Name))
+		return nil, fmt.Errorf("repo not set for req name:%s", req.Name)
+	}
+	hash := req.GetParameters()["release"]
+	if hash == "" {
+		s.config.driver.config.Logger.Error("Release not set", zap.String("name", req.Name))
+		return nil, fmt.Errorf("release not set for req name:%s", req.Name)
+	}
+	rd, err := core.GetRepo(repo, s.config.driver.metadataStore)
```

Comment:
so `CreateVolume` creates a volume corresponding to an _existing_ repo...

Reply:
Yes, it is not a create for the backend but a create operation for the reference in kubernetes for a volume external to it.